### ISSUE TITLE
repl multi line statements not working

### ISSUE THREAD

**Issue by dsl400** [Carglglz/upydev#34]
shell-repl @ upydev
WebREPL connected
WARNING: ENCRYPTION DISABLED IN THIS MODE

MicroPython v1.18 on 2022-01-17; ESP module (1M) with ESP8266
Type "help()" for more information.

- CTRL-k to see keybindings or -h to see help
- CTRL-s to toggle shell/repl mode
- CTRL-x or "exit" to exit
esp8266@temp1:~ $ 
>>> for i in [1,2]:
        print(i)

............
....

no key combination will get you out of this situation
you have to close the terminal and start another one

**Comment by Carglglz** [comment]
@dsl400 

Yes this is a known limitation see [shell-repl](https://upydev.readthedocs.io/en/latest/shell_repl.html)

> The REPL mode has two limitations:

> It is not listening actively for output (This means that if a timer/hardware interrupt callback print something it will not appear in the repl).
To define a function/class or make a loop use the paste mode. (CTRL-E)

> However the original WebREPL/Serial REPL can be accessed from shell with repl command

Although  I may have to update this to clarify that this 
> To define a function/class or make a loop use the paste mode. (CTRL-E)

means this too 

> repl multi line statements not working

See also [Device development setups](https://upydev.readthedocs.io/en/latest/examples.html#serialdevice) which may be of help. 👍🏼

**Comment by dsl400** [comment]
> The easiest way to develop is having the device directly connected to the computer by USB

yeah .... but ... it is not always possible to stay connected via usb / serial 
![image](https://user-images.githubusercontent.com/4988046/197065549-1d6886d7-b13b-4e6b-b58a-c777089b05ea.png)
![image](https://user-images.githubusercontent.com/4988046/197065602-d39a7ea1-ba63-4236-b8c9-32f403138a7f.png)

I'm using upydev because it can upload files via wifi with ease

**Comment by Carglglz** [comment]
> it is not always possible to stay connected via usb / serial

Yes, or if you have a bunch of devices in different places around the house, building, or in a remote location... that's why I've developed  `upydev` and also note that this [Device development setups](https://upydev.readthedocs.io/en/latest/examples.html#serialdevice) applies to `WebSocketDevices` too.

Also if you are using `esp8266` you may be insterested in freezing scipts with `mpy-cross` so it saves some precious RAM.
